tests/integration/memory_resource_traits_tests.cpp

```diff
@@ -7,129 +7,104 @@
-
-TEST(MemoryResourceTraitsTest, HOST_Resource)
+//
+//This test confirms that when an allocator is created with a specific
+//memory resource, that same memory resource can be quiered and returned
+//
+class MemoryResourceTraitsTest : public ::testing::TestWithParam<std::string> {
+ public:
+  virtual void SetUp()
+  {
+    auto& rm = umpire::ResourceManager::getInstance();
+    
+    m_allocator = new umpire::Allocator(rm.getAllocator(GetParam()));
+    m_allocator_pool = new umpire::Allocator(rm.makeAllocator<umpire::strategy::QuickPool>
+        ("pool_" + GetParam(), *m_allocator));
+    
+    m_resource = GetParam();
+  }
+
+  virtual void TearDown()
+  {
+    if(m_allocator)
+      delete m_allocator;
+    if(m_allocator_pool)
+      delete m_allocator_pool;
+  }
+
+  umpire::Allocator* m_allocator;
+  umpire::Allocator* m_allocator_pool;
+  std::string m_resource;
+};
+
+umpire::MemoryResourceTraits::resource_type get_resource_trait(std::string resource)
 {
-  auto& rm = umpire::ResourceManager::getInstance();
-
-  auto alloc_one = rm.getAllocator("HOST");
-  auto alloc_two = rm.makeAllocator<umpire::strategy::SizeLimiter>(
-      "HOST_Limiter", alloc_one, 1024);
-
-  double* data =
-      static_cast<double*>(alloc_one.allocate(1024 * sizeof(double)));
-
-  ASSERT_EQ(MemoryResourceTraits::resource_type::host,
-            alloc_two.getAllocationStrategy()->getTraits().resource);
-
-  ASSERT_THROW(alloc_two.deallocate(data), umpire::util::Exception);
-
-  ASSERT_NO_THROW(alloc_one.deallocate(data));
+  if(resource == "HOST")
+    return umpire::MemoryResourceTraits::resource_type::host;
+  else if(resource.find("::") != std::string::npos ||
+          resource == "DEVICE")
+    return umpire::MemoryResourceTraits::resource_type::device;
+  else if(resource == "DEVICE_CONST")
+    return umpire::MemoryResourceTraits::resource_type::device_const;
+  else if(resource == "UM")
+    return umpire::MemoryResourceTraits::resource_type::um;
+  else if(resource == "PINNED")
+    return umpire::MemoryResourceTraits::resource_type::pinned;
+  else if(resource == "FILE")
+    return umpire::MemoryResourceTraits::resource_type::file;
+  else
+    return umpire::MemoryResourceTraits::resource_type::unknown;
 }
 
-#if defined(UMPIRE_ENABLE_FILE_RESOURCE)
-TEST(MemoryResourceTraitsTest, FILE_Resource)
+TEST_P(MemoryResourceTraitsTest, ResourceTraitTest)
 {
-  auto& rm = umpire::ResourceManager::getInstance();
+  double* data =
+      static_cast<double*>(m_allocator->allocate(1024 * sizeof(double)));
 
-  auto alloc_one = rm.getAllocator("FILE");
-  auto alloc_two = rm.makeAllocator<umpire::strategy::SizeLimiter>(
-      "FILE_Limiter", alloc_one, 1024);
+  umpire::MemoryResourceTraits::resource_type resource = get_resource_trait(m_resource);
 
-  double* data =
-      static_cast<double*>(alloc_one.allocate(1024 * sizeof(double)));
+  ASSERT_EQ(resource, m_allocator_pool->getAllocationStrategy()->getTraits().resource);
```

Comment:
Can you check the getTraits().resource for m_allocator as well.

Reply:
Si señor